storage/s3.go

```diff
@@ -347,6 +348,14 @@ func (s *S3) Copy(ctx context.Context, from, to *url.URL, metadata Metadata) err
 	return err
 }
 
+func escapeKey(sourceKey string) string {
+	sourceKeyElements := regexp.MustCompile(`/`).Split(sourceKey, -1)
```

Comment:
why not `strings.Split(sourceKey, "/")` ?

and could you please add a unit test for this function.

Reply:
I intended for other special-characters but it is not necessary because they could be either in:
- safe group: no need to handle, except `/`
- should avoid group.

I will update it as your suggestion